chore(node/buffer): convert to js private variables

sno2 opened the pull request:
See #1649 [please wait for CI]

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/denoland/deno_std?pullRequest=1655) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/denoland/deno_std?pullRequest=1655) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/denoland/deno_std?pullRequest=1655) it.</sub>

sno2 commented:
I'm sorry @bartlomieju , but I can't seem to find out why the tests are hanging.  All I did was change the `BufferReader` abstract class to not unsafe anymore and rename all "private" variable / method instances to `#__`.

kt3k commented:
@sno2 
This seems crashing even without rust panic (maybe it crashes inside of V8?). The crash seems happening around `#reset` method of `BufReader`, but I couldn't figure out what exactly causes this.

If I renamed `#reset` (of BufReader) to `_reset`, then the issue disappeared on my local machine.

sno2 commented:
> @sno2 This seems crashing even without rust panic (maybe it crashes inside of V8?). The crash seems happening around `#reset` method of `BufReader`, but I couldn't figure out what exactly causes this.
> 
> If I renamed `#reset` (of BufReader) to `_reset`, then the issue disappeared on my local machine.

Yeah, ummm when I changed `#reset` to an arrow function it passed.  I am so confused because I have no idea why it would behave differently.  We'll see what CI says, though.

kt3k commented on io/buffer.ts:
```diff
@@ -293,40 +293,40 @@ export class BufReader implements Reader {
 
   // Reads a new chunk into the buffer.
-  private async _fill() {
+  async #fill() {
```
Regarding the case of `#reset`, I suspect this might also cause the same issue..

kt3k commented:
Otherwise looks good to me

kt3k reviewed (approved):
LGTM. Nice work!
